src/controllers/skills/answerFactory.ts

```diff
@@ -0,0 +1,62 @@
+import {Answer} from './answer'
+
+export class AnswerFactory {
+	constructor() {
+		this.create = this.create.bind(this)
+	}
+
+	create(data: any) {
+		const answer = new Answer()
+		let answers: any = {}
+
+		let charCount = 65
+		for (let i = 0; i < 5; i++) {
+			console.log(charCount + " : "+ String.fromCharCode(charCount))
+			if (Array.isArray(data.answers) && data.answers[0] && data.answers[i] != "") {
+				// @ts-ignore
+				answers[String.fromCharCode(charCount)] = data.answers[i]
+				charCount++
+			} else if(!Array.isArray(data.answers)) {
+				// @ts-ignore
+				answers[String.fromCharCode(charCount)] = this.getKeyCode(i,data)
+				charCount++
+			}
+		}
+		if(data.id && data.id != "") {
+			answer.id = data.id
+		}
+		answer.answers = answers
+		if(Array.isArray(data.correctAnswers)) {
+			answer.correctAnswers = data.correctAnswers
+		} else {
+			let array = []
+			array.push(data.correctAnswers)
+			answer.correctAnswers = array
+			console.log("wroks")
+		}
+
+		return answer
+	}
+
+	getKeyCode(id: any, data: any) {
+		let value: any = ""
+		switch(id) {
```

Comment:
Do you allow selection of anything after D

Reply:
good point, I added E as well. E will be the last one as admin can only add questions with max of 5 answers